bftengine/src/preprocessor/PreProcessor.cpp

```diff
@@ -1537,7 +1545,10 @@ void PreProcessor::releaseClientPreProcessRequest(const RequestStateSharedPtr &r
       givenReq.reset();
     }
     if (batchedPreProcessEnabled_ && !batchCid.empty()) ongoingReqBatches_[clientId]->increaseNumOfCompletedReqs(1);
-    if (!myReplica_.isCurrentPrimary()) {
+    /* we added the below preProcInFlyRequestsNum >0 check to prevent the buffer overflow due to more decrements during
```

Comment:
Please replace the comment with "Prevent preProcInFlyRequestsNum metric to be overflowed during a view change"

Reply:
sure